torch/_inductor/fx_passes/quantization.py

```diff
@@ -1071,9 +1071,15 @@ def _register_quantization_reshape():
 def _is_valid_woq_optimization_pattern():
     def fn(match):
         assert all(k in match.kwargs for k in ("x", "weight", "scales"))
-        x = match.kwargs["x"].meta["val"]
-        weight = match.kwargs["weight"].meta["val"]
-        scales = match.kwargs["scales"].meta["val"]
+        x = match.kwargs["x"]
+        if hasattr(x, 'meta'):
+            x = x.meta["val"]
+        weight = match.kwargs["weight"]
+        if hasattr(weight, 'meta'):
+            weight = weight.meta["val"]
+        scales = match.kwargs["scales"]
+        if hasattr(scales, 'meta'):
+            scales = scales.meta["val"]
```

Comment:
Thanks for the PR! If any of the `x/weight/scales` cannot retrieve `meta["val"]`, this function should return `False`, as we cannot guarantee the conditions below.

Reply:
thanks for pointing that out!

Reply:
returning false now